core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/console-reporter.ts

```diff
@@ -0,0 +1,184 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility File for detecting, filtering and reporting
+ * certain console logs while running puppeteer tests.
+ */
+
+import {
+  ConsoleMessage as PuppeteerConsoleMessage,
+  Browser,
+  Target,
+  ConsoleMessageType,
+  JSHandle,
+} from 'puppeteer';
+
+import escapeRegExp from 'lodash/escapeRegExp';
+
+const HOST_URL = 'http://localhost:8181';
+
+interface ConsoleMessage {
+  type: ConsoleMessageType;
+  text: string;
+  url: string;
+}
+
+const CONSOLE_ERRORS_TO_IGNORE = [
+  // These "localhost:9099" are errors related to communicating with the
+  // Firebase emulator, which would never occur in production, so we just ignore
+  // them.
+  escapeRegExp(
+    'http://localhost:9099/www.googleapis.com/identitytoolkit/v3/' +
+      'relyingparty/getAccountInfo?key=fake-api-key'
+  ),
+  escapeRegExp(
+    'http://localhost:9099/www.googleapis.com/identitytoolkit/v3/' +
+      'relyingparty/verifyPassword?key=fake-api-key'
+  ),
+  // This error covers the case when the PencilCode site uses an
+  // invalid SSL certificate (which can happen when it expires).
+  // In such cases, we ignore the error since it is out of our control.
+  escapeRegExp(
+    'https://pencilcode.net/lib/pencilcodeembed.js - Failed to ' +
+      'load resource: net::ERR_CERT_DATE_INVALID'
+  ),
+  // These errors are related to the gtag script that is used to track events.
+  // They are of the form "Failed to load resource: the server responded
+  // with a status of 405", this happens when the HTTP method used for a
+  // network call is refused by the server. The network call is triggered
+  // automatically by the gtag script, so we have no control over it. The 405
+  // error was observed on other websites (e.g. https://edu.google.com/) that
+  // use gtag and it does not affect the user experience in any way.
+  // Considering these reasons, the error may be ignored.
+  new RegExp(
+    'https://www.googletagmanager.com/a.* Failed to load resource: ' +
+      'the server responded with a status of 405 ()',
+    'g'
+  ),
+];
+
+const CONSOLE_ERRORS_TO_FIX = [
+  // TODO(#19746): Development console error "Uncaught in Promise" on signup.
+  new RegExp(
+    'Uncaught \\(in promise\\).*learner_groups_feature_status_handler'
+  ),
+  // TODO(#19687): 404 (Not Found) for favicon on home page.
```

Comment:
This seems closed, is it still actually occurring?

Reply:
Hm... tested on local and it doesn't show up but on acceptance tests it does.

Reply:
Hm, that's odd. It might be worth investigating why that happens ... might indicate a larger bug (it's quite important that we don't have skew between local and the acceptance tests).

Reply:
It only happens on prod env so should reopen issue and reclarify that it only happens in prod (--prod_env flag inside start)

Reply:
Ah ok, please could you go ahead and do that (feel free to add any necessary detail)? Or it's OK to file an issue and link to the old one for context.

Thanks!

Reply:
Done